### ISSUE TITLE
SBML Model not supporting time series

### ISSUE THREAD

**Issue by kaiserls** [Systems-Theory-in-Systems-Biology/EPI#71]
**Describe the bug**
SBML models returns the solution at `time[0]` instead of the solution at all timepoints in `time`.

**To Reproduce**
```python
import numpy as np
import importlib
from eulerpi.core.model import SBMLModel
from eulerpi.core.inference import inference

CENTRAL_PARAM = np.array([1.0, 1.0])
PARAM_LIMITS = np.array([[0.0, 2.0], [0.0, 2.0]])

sbml_file = importlib.resources.path(
        "eulerpi.examples.sbml", "Caffeine_2Wks_Exponential_decay.xml"
    )

model = SBMLModel(sbml_file=sbml_file, time=[1.0], central_param=CENTRAL_PARAM, param_limits=PARAM_LIMITS, skip_creation=True)
print(model.forward(CENTRAL_PARAM).shape)
model = SBMLModel(sbml_file=sbml_file, time=[0.5, 1.0], central_param=CENTRAL_PARAM, param_limits=PARAM_LIMITS, skip_creation=True)
print(model.forward(CENTRAL_PARAM).shape)
```

**Expected behavior**
The sbml model returns the solution at all timepoints in `time` as flat numpy array.

**Observed behavior**
The sbml model returns the solution at the timepoint `time[0]`.

**Comment by shoepfl** [comment]
Hi Lars,

thanks for opening the Issue - I currently try to get is working for one species and arbitrary timepoints.
The simulation results and jacobian from SBML are now as expected in the SBML feature branch but there are still some issues to fix.

**Comment by kaiserls** [comment]
Hi Sebastian,

thank you for the quick response and your work on the sbml model! If you want to merge a fully working sbml example into the main branch I would recommend that you activate the tests for one of the sbml example models. You can do this by removing the xfail mark for one of the sbml models in` tests/test_examples.py`. This will allow you to run many tests for the model, e.g. some that confirm that the shapes of the arrays returned by the forward and jacobian function are compatible with the rest of the epi code. You can run the tests locally with `poetry run pytest -v -k sbml`.

I think taking the full arrays x / sx and reshaping them into (n_species * n_timepoints) or (n_species * n_timepoints, n_params) for the forward and jacobian should be a solid basis. It would be come with the need to modify the data_dim property to something like the following:
```
@property
def data_dim(self):
   """The number of observables of the model."""
   return len(self.amici_model.getStateIds()) * len(self.time)

```

The next step, would be dealing with linearly dependent species concentrations like in the menten model. But this can be the job of the user, we should just fix it in our example model.

**Comment by kaiserls** [comment]
I took a closer look at your changes and like the change to the parameter ids from the parameter names and obviously the support for time series. Would you welcome commits from my side regarding the building of the jacobian and parameter and state selection?

**Comment by shoepfl** [comment]
Hi Lars,

as you might have seen I first tried to support time series for one species only. However, of course the final goal before merging this Branch back to main would be to support also more than one species. So reshaping according to `n_species * n_timepoints` is a good idea I would say.

Of course you are very welcome to commit :) I have also only limited time to work on this at the moment as there are two conferences next week and the week after. So just go ahead and when I go on working I will first update my SBML branch
